src/coreclr/src/jit/codegenxarch.cpp

```diff
@@ -7491,6 +7464,44 @@ void CodeGen::genBitCast(var_types targetType, regNumber targetReg, var_types sr
     }
 }
 
+//----------------------------------------------------------------------
+// genCodeForBitCast - Generate code for a GT_BITCAST that is not contained
+//
+// Arguments
+//    treeNode - the GT_BITCAST for which we're generating code
+//
+void CodeGen::genCodeForBitCast(GenTreeOp* treeNode)
+{
+    regNumber targetReg  = treeNode->GetRegNum();
+    var_types targetType = treeNode->TypeGet();
+    GenTree*  op1        = treeNode->gtGetOp1();
+    genConsumeRegs(op1);
+
+    if (op1->isContained())
+    {
+        assert(op1->IsLocal() || op1->isIndir());
+        if (genIsRegCandidateLocal(op1))
+        {
+            unsigned lclNum = op1->AsLclVar()->GetLclNum();
+            GetEmitter()->emitIns_R_S(ins_Load(treeNode->TypeGet(), compiler->isSIMDTypeLocalAligned(lclNum)),
+                                      emitTypeSize(treeNode), targetReg, lclNum, 0);
+        }
+        else
+        {
+            op1->gtType = treeNode->TypeGet();
```

Comment:
Changing IR nodes during codegen, odd...

Reply:
Yes, it's a bit weird. and one might consider changing it during `Lowering`, except that we can wind up here when the source is a register candidate lclVar that winds up being in memory here. It will be marked `RegOptional` and we don't want to retype it if it actually gets a register.
I'm open to better ways of handling this. Note that we're not currently using `BITCAST` in all the places that we will eventually want to, such as outgoing args, so in future there will be more cases.

Reply:
> except that we can wind up here when the source is a register candidate lclVar that winds up being in memory here

At least the indir case could be handled by retyping in lowering. I'm not sure what's going on in the case of a non-reg candidate lcl, if it's not reg candidate then it is in memory so you can simply load it using `emitIns_R_S`, similar what the existing code already does.

Reply:
The case where it's tricky is where it's a register candidate, but has been spilled. In that case we don't want to retype the lclvar node in `Lowering`, because it may have a register, and we want to keep it in its original register file and do a copy. In the case where it's been spilled and is marked `RegOptional` that's the case where we want to retype the node so that its parent can load it directly into the desired register file.